amazon_transcribe/serialize.py

```diff
@@ -12,75 +12,114 @@
-        raise NotImplementedError("serialize")
-
-    def serialize_to_request(self) -> PreparedRequest:
-        """Serialize parameters into an HTTP request."""
-        raise NotImplementedError("serialize_to_request")
-
-
-class TranscribeStreamingRequestSerializer(Serializer):
+class TranscribeStreamingSerializer:
     """Convert StartStreamTranscriptionRequest into a
     Request object for streaming to the Transcribe service.
     """
 
-    def __init__(self, endpoint, transcribe_request):
-        self.endpoint: str = endpoint
-        self.method: str = "POST"
-        self.request_uri: str = "/stream-transcription"
-        self.request_shape: StartStreamTranscriptionRequest = transcribe_request
+    def _raise_param_validation_exception(self, value, header, expected):
+        raise ValidationException(
+            f"Unexpected param type ({type(value)}) encountered for {header}, expected type: {expected}"
+        )
 
-    def serialize(self) -> Tuple[Dict[str, HEADER_VALUE], BufferedIOBase]:
-        headers = {
-            "x-amzn-transcribe-language-code": self.request_shape.language_code,
-            "x-amzn-transcribe-sample-rate": self.request_shape.media_sample_rate_hz,
-            "x-amzn-transcribe-media-encoding": self.request_shape.media_encoding,
-            "x-amzn-transcribe-vocabulary-name": self.request_shape.vocabulary_name,
-            "x-amzn-transcribe-session-id": self.request_shape.session_id,
-            "x-amzn-transcribe-vocabulary-filter-method": self.request_shape.vocab_filter_method,
-            "x-amzn-transcribe-vocabulary-filter-name": self.request_shape.vocab_filter_name,
-            "x-amzn-transcribe-show-speaker-label": self.request_shape.show_speaker_label,
-            "x-amzn-transcribe-enable-channel-identification": self.request_shape.enable_channel_identification,
-            "x-amzn-transcribe-number-of-channels": self.request_shape.number_of_channels,
-        }
+    def _serialize_header(
+        self, header: str, expected_type, value, prefix: str = "x-amzn-transcribe-"
+    ):
+        if value is None:
+            return {}
+        elif not isinstance(value, expected_type):
+            self._raise_param_validation_exception(
+                value, header, expected_type.__name__
+            )
+        else:
+            return {f"{prefix}{header}": str(value)}
+
+    def _serialize_str_header(self, header: str, value: Optional[str]):
+        return self._serialize_header(header, str, value)
+
+    def _serialize_int_header(self, header: str, value: Optional[int]):
+        return self._serialize_header(header, int, value)
+
+    def _serialize_bool_header(self, header: str, value: Optional[bool]):
+        return self._serialize_header(header, bool, value)
+
+    def serialize_start_stream_transcription_request(
+        self, endpoint: str, request_shape: StartStreamTranscriptionRequest
+    ) -> Request:
+
+        method: str = "POST"
```

Comment:
Do these need to be typed since they're function scope and not supplied from external input?